__tests__/cake.test.ts

```diff
@@ -190,3 +193,58 @@ describe('When running a script successfully using the tool manifest', () => {
       ['script.cake', '--switch']);
   });
 });
+
+describe('When running a Cake Frosting project successfully', () => {
+  const fakeExec = exec as jest.MockedFunction<typeof exec>;
+  const fakeToolsDirectory = new ToolsDirectory();
+
+  beforeAll(async () => {
+    fakeExec.mockReturnValue(Promise.resolve(0));
+  });
+
+  test('it should run with the default non-required parameters', async () => {
+    await cake.runProject(pathToCsprojFile, fakeToolsDirectory);
+
+    expect(fakeExec).toHaveBeenCalledWith(
+      dotnetRun,
+      [
+        '--project', pathToCsprojFile,
+        '--no-launch-profile',
+        '--verbosity', 'minimal',
+        '--configuration', 'Release',
+        `--paths_tools="${fakeToolsDirectory.path}"`
+      ]);
+  });
+
+  test('it should run with the specified parameters', async () => {
+    await cake.runProject(
+      pathToCsprojFile,
+      fakeToolsDirectory,
+      new CakeArgument('param', 'arg'),
+      new CakeSwitch('switch'));
+
+    expect(fakeExec).toHaveBeenCalledWith(
+      dotnetRun,
+      [
+        '--project', pathToCsprojFile,
+        '--no-launch-profile',
+        '--verbosity', 'minimal',
+        '--configuration', 'Release',
+        `--paths_tools="${fakeToolsDirectory.path}"`,
+        '--param=arg',
+        '--switch'
+      ]);
+  });
+});
+
+describe('When failing to run a Cake Frosting Project', () => {
+  const fakeExec = exec as jest.MockedFunction<typeof exec>;
+
+  beforeAll(() => {
+    fakeExec.mockReturnValue(Promise.resolve(-21));
+  });
+
+  test('it should throw error when csproj-path does not exist', async () => {
```

Comment:
**issue:** the test implementation doesn't reflect this requirement, since the `exec` function is stubbed to always return `-21`.

```suggestion
  test('it should throw an error containing the exit code', async () => {
```